src/re-rxjs/index.js

```diff
@@ -0,0 +1,79 @@
+import React, { unstable_useTransition as useTransition } from 'react';
+import { Subject, merge, asapScheduler } from 'rxjs';
+import {
+  map, scan, startWith, observeOn,
+} from 'rxjs/operators';
+import { connectObservable } from 're-rxjs';
+import {
+  syncBlock,
+  useRegisterIncrementDispatcher,
+  ids,
+  useCheckTearing,
+} from '../common';
+
+const normalClicks$ = new Subject();
+const normalIncrement = () => normalClicks$.next();
+
+const externalClicks$ = new Subject();
+// This is only needed to prevent the tearing with auto increment (check #9)
+const externalClicksAsap$ = externalClicks$.pipe(observeOn(asapScheduler));
+const externalIncrement = () => externalClicks$.next();
+
+const [useCounter] = connectObservable(
+  merge(normalClicks$, externalClicksAsap$).pipe(
+    scan((x) => x + 0.5, 0),
+    map(Math.floor),
+    startWith(0),
+  ),
+);
+
+const Counter = React.memo(() => {
+  const count = useCounter();
+  syncBlock();
+  return <div className="count">{count}</div>;
+});
+
+const Main = () => {
+  const count = useCounter();
+  useCheckTearing();
+  useRegisterIncrementDispatcher(externalIncrement);
+  const [localCount, localIncrement] = React.useReducer((c) => c + 1, 0);
+  const [startTransition, isPending] = useTransition();
+  const transitionIncrement = () => {
+    startTransition(() => {
+      normalIncrement();
+    });
+  };
+
+  return (
+    <div>
+      <button
+        type="button"
+        id="normalIncrement"
+        onClick={isPending ? transitionIncrement : normalIncrement}
```

Comment:
This is weird. Just do this:
```suggestion
        onClick={normalIncrement}
```